### PR TITLE
feat: support more element types

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -36,12 +37,6 @@ export default {
     resolve({
       mainFields: ['module', 'main', 'jsnext', 'browser'],
     }),
-    commonjs({
-      sourceMap: true,
-      namedExports: {
-        react: ['isValidElement'],
-      },
-    }),
```
Comment (armandabric) [inline rollup.config.js:44]: Can you explain why you have to change that in the context of this PR?

**Review by armandabric — COMMENTED** [review]
Wow! Thanks for this really god PR. I just have a question regarding the rollup configuration part.

File: rollup.config.js (hunk 1 above)
Comment (Andarist) [reply rollup.config.js:44]: `commonjs` plugin is only needed when you actually want to **bundle** a commonjs dependency - you don't really want that with `react` since you are leaving it as external (like you should), so this was actually a "dead"/redundant config

**Comment by armandabric** [comment]
This PR will fix #352 and is related to #617

**Comment by Andarist** [comment]
@armandabric I've fixed the Flow issue 🎉  Also - I think that you have wanted to link another issue/PR since #617 is this one

File: src/parser/parseReactElement.js [hunk 2]
```diff
@@ -12,12 +24,58 @@ import type { TreeNode } from './../tree';
...
+      return 'Lazy';
+    case isProfiler(element):
+      return 'Profiler';
+    case isStrictMode(element):
+      return 'StrictMode';
+    case isSuspense(element):
+      return 'Suspense';
+    default:
+      // this is "unknown" case and will most likely result in `[object Object]`
+      // it's fine though as we can't display anything meaningful anyway here
```
Comment (Andarist) [inline src/parser/parseReactElement.js:75]: Let me know if you prefer returning `"UnknownElementType"` here or something.

Comment (armandabric) [reply src/parser/parseReactElement.js:75]: I like your idea to return `UnknownElementType`. It will help to debug issue.

File: rollup.config.js (hunk 1 above)
Comment (armandabric) [reply rollup.config.js:44]: Let's improve this!

Comment (Andarist) [reply rollup.config.js:44]: Hm, could you be more specific? maybe I've explained this poorly - so let me rephrase: the removed code that was not actually used and it wasn't doing anything here. And how this config now is defined is correct - from my point of view this change is "sound" and correct. There is no regression here so this can be merged "as is". Please let me know if this addresses your concerns - if not I can try to explain it further and in more detail.

Comment (armandabric) [reply rollup.config.js:44]: Ho! Sorry I just wanted to said _Let's change this!_ in the meaning of accepting your change with pleasure. Sorry for the wrong wording :smiling_face_with_tear:

**Review by armandabric — APPROVED** [review]
Thanks again for this PR! That was a long time running issue for the library :ok_hand:

**Comment by xyy94813** [comment]
For `@storybook/addon-docs`, the scenario of converting a lazy component to a string is a little different from the needs of `react-dev-tools`.

```js
function MyComponent() {
  return 
}

const LazyComponent = React.lazy(async () => {
  return MyComponent
});

const root = React.render(
  <LazyComponent />
);

// for react-dev-tools
expect(jsxToString(root)).toBe(<Lazy />);  // first
expect(jsxToString(root)).toBe(<MyComponent />); // then

// for `@storybook/addon-docs` i will except
expect(jsxToString(root)).toBe(<Lazy(MyComponent) />); // then
```

Any suggestions about element to Source code?

**Comment by Andarist** [comment]
Ah yes - I was wondering about this case but the problem here is that this `LazyComponent` doesn't have a name at all. We could try to infer it from the inner function **but** that is not an established pattern for `React.lazy`. We can't easily access the `MyComponent` name in the given example during the runtime.
